packages/terra-theme-provider/src/ThemeProvider.jsx

```diff
@@ -18,52 +15,27 @@ const propTypes = {
    * Name of class for specified theme. e.g `cerner-consumer-theme`
    */
   themeName: PropTypes.string,
-  /**
-   * When set to true, applies theme class to HTML element
-   */
-  isGlobalTheme: PropTypes.bool,
 };
 
-const defaultProps = {
-  isGlobalTheme: false,
-};
-
-class ThemeProvider extends React.Component {
-  componentDidMount() {
-    if (this.props.isGlobalTheme === true && this.props.themeName) {
-      document.documentElement.classList.add(this.props.themeName);
+const ThemeProvider = ({
+  children,
+  themeName,
+}) => {
+  useEffect(() => {
+    if (themeName) {
+      document.documentElement.classList.add(themeName);
```

Comment:
You should add a ref for the previous theme name and remove it before adding the current theme name (if they are different).

Reply:
As per my current implementation, on switching from one theme to another, the previously applied theme will be removed first before the new theme is added.
To implement this i used the following as [reference](https://reactjs.org/docs/hooks-effect.html#explanation-why-effects-run-on-each-update).
Also, this will only execute when the theme name has been changed : [reference](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).

The only problem that i am facing with the current approach is that i am getting an exception from PropsTable (only on the main Documentation page). Besides that all the other functionality seems to be the same

**I am not sure if i missed something, but is there a benefit of using the ref approach over this?**

Reply:
@avinashg1994 You are correct, you don't need a ref, which does make things a lot cleaner.